src/systemctl/systemctl-list-units.c

```diff
@@ -105,11 +106,200 @@ static char *format_unit_id(const char *unit, const char *machine) {
         return machine ? strjoin(machine, ":", unit) : strdup(unit);
 }
 
-static int output_units_list(const UnitInfo *unit_infos, size_t c) {
+static Hashmap *slice_unit_map = NULL;
+typedef struct {
+        char ** units;
+        size_t num_units;
+}SliceUnits;
```

Comment:
just use a strv, see src/basic/strv.h

Reply:
Thank you for your review sir @bluca, for this case i used a struct instead of strv because both the `hashmap_put` and `hashmap_get` both take a pointer and return a pointer respectively. 
`static inline void *hashmap_get(Hashmap *h, const void *key)` and 
`static inline int hashmap_put(Hashmap *h, const void *key, void *value) `
However the units under a slice is an array of pointers to strings(char **). so that is why i used the struct to just pass the functions its pointer.